src/Microsoft.Health.Fhir.R4.Core/Features/Resources/Upsert/UpsertResourceHandler.cs

```diff
@@ -0,0 +1,24 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Threading;
+using System.Threading.Tasks;
+using MediatR;
+using Microsoft.Health.Fhir.Core.Features.Persistence;
+using Microsoft.Health.Fhir.Core.Messages.Upsert;
+
+namespace Microsoft.Health.Fhir.Core.Features.Resources.Upsert
+{
+    public partial class UpsertResourceHandler : BaseResourceHandler, IRequestHandler<UpsertResourceRequest, UpsertResourceResponse>
+    {
+        private async Task<UpsertOutcome> UpsertAsync(UpsertResourceRequest message, ResourceWrapper resourceWrapper, bool allowCreate, bool keepHistory, CancellationToken cancellationToken)
+        {
+            // In R4, the server returns a 412 Precondition Failed status code if the version id given in the If-Match header does not match
```

Comment:
This comment makes sense in the context of this code review, but seems out of place right here in the code. We're not doing anything special for this scenario in the code that follows.